src/Payum/Core/Bridge/PlainPhp/Security/TokenFactory.php

```diff
@@ -31,7 +31,17 @@ public function __construct(StorageInterface $tokenStorage, StorageRegistryInter
      */
     protected function generateUrl($path, array $parameters = [])
     {
-        $newPath = (new Path($this->baseUrl->getPath()))->withTrailingSlash().$path;
```

Comment:
I think this can be simplified to one line, then you don't need all the extra checks below:

```php
$newPath = UriModifier::replaceBasename($this->baseUrl, $path)->getPath();
```

@detain Do you mind making the change, and also adding a unit test?